pkg/monitoring/pb/data-feeds/on-chain/registry/decode.go

```diff
@@ -5,25 +5,22 @@ import (
 	"math"
 	"math/big"
 
-	wt_msg "github.com/smartcontractkit/chainlink-evm/pkg/report/pb/platform"
+	wt_msg "github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/monitoring/pb/platform"
+	"github.com/smartcontractkit/chainlink-framework/capabilities/writetarget/report/platform"
 
 	"github.com/smartcontractkit/chainlink-evm/pkg/report/datafeeds"
-	"github.com/smartcontractkit/chainlink-evm/pkg/report/platform"
-
-	mercury_vX "github.com/smartcontractkit/chainlink-evm/pkg/report/mercury/common"
-	mercury_v3 "github.com/smartcontractkit/chainlink-evm/pkg/report/mercury/v3"
-	mercury_v4 "github.com/smartcontractkit/chainlink-evm/pkg/report/mercury/v4"
+	"github.com/smartcontractkit/chainlink-evm/pkg/report/por"
 )
 
-func DecodeAsFeedUpdated(m *wt_msg.WriteConfirmed) ([]*FeedUpdated, error) {
+func DecodeAsFeedUpdated(m *wt_msg.WriteConfirmed, ccip bool) ([]*FeedUpdated, error) {
```

Comment:
I'm don't particularly like this ccip bool flag on the input, and that these two fn are completely the same (`DecodeAsFeedUpdated` & `DecodePORAsFeedUpdated`), except the underlying report decoding step.

It would be best to consider extracting a higher-level fn out, and injecting just the underlying report decoding step.